dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -387,50 +388,38 @@ def _update_histogram(self, df_series):
             current_total_var, current_run_time)
         self.histogram_selection = selected_method
 
-    def _get_percentile(self, percentile):
+    def _get_percentile(self, percentiles):
         """
         Get value for the number where the given percentage of values fall below
         it.
 
-        :param percentile: Percentage of values to fall before the value
-        :type percentile: float
-        :return: Value for which the percentage of values in the distribution
-            fall before the percentage
+        :param percentiles: List of percentage of values to fall before the
+            value
+        :type percentiles: list[float]
+        :return: List of corresponding values for which the percentage of values
+            in the distribution fall before each percentage
         """
+
         selected_method = self.histogram_selection
         bin_counts = \
             self.histogram_methods[selected_method]['histogram']['bin_counts']
         bin_edges = \
             self.histogram_methods[selected_method]['histogram']['bin_edges']
-        num_edges = len(bin_edges)
 
-        if percentile == 100:
-            return bin_edges[-1]
-        percentile = float(percentile) / 100
+        non_zero_inds = bin_counts == 0
 
-        accumulated_count = 0
         bin_counts = bin_counts.astype(float)
         normalized_bin_counts = bin_counts / np.sum(bin_counts)
+        cumsum_bin_counts = np.cumsum(normalized_bin_counts)
 
-        bin_id = -1
-        # keep updating the total counts until it is
-        # close to the designated percentile
-        while accumulated_count < percentile:
-            bin_id += 1
-            accumulated_count += normalized_bin_counts[bin_id]
+        # use the floor by slightly increasing cases where no bin exist.
+        cumsum_bin_counts[non_zero_inds] += 1e-12
```

Comment:
interp will choose the higher of the 2 values, so a small addition is added to prevent it.